packages/core/src/internal/proc.js

```diff
@@ -150,6 +150,7 @@ function createTaskIterator({ context, fn, args }) {
 
 export default function proc(iterator, parentContext, parentEffectId, meta, env, cont) {
   const taskContext = Object.create(parentContext)
+  const finalRunEffect = env.finalizeRunEffect(runEffect)
```

Comment:
Maybe `enhancedRunEffect` and `env.enhanceRunEffect` are better names. I don't like marking things as finalized, since we will change them again someday.